packages/headless/src/features/folding/folding-slice.ts

```diff
@@ -37,70 +42,113 @@ function areDefinedAndEqual<T>(
   return (value1 || value2) !== undefined && value1 === value2;
 }
 
-function flattenResultTree(result: ResultWithFolding): ResultWithFolding[] {
-  const results = [result];
-  if (result.parentResult) {
-    results.unshift(result.parentResult);
-  }
-  if (result.childResults?.length) {
-    results.push(
-      ...result.childResults.flatMap((childResult) =>
-        flattenResultTree(childResult)
-      )
-    );
+function getFoldedResults(result: ResultWithFolding): ResultWithFolding[] {
+  if (!result.childResults) {
+    return [];
   }
-  return results;
+  return result.childResults.flatMap((childResult) => [
+    childResult,
+    ...getFoldedResults(childResult),
+  ]);
 }
```

Comment:
I'm going to update the other PR with this change. I found that it's not necessary to flatten all results, and I felt like "getFoldedResults" was a more accurate terminology since this corresponds with `numberOfFoldedResults`.